packages/amplify_authenticator/lib/src/widgets/authenticator_banner.dart

```diff
@@ -21,60 +21,124 @@ import 'package:flutter/material.dart';
 MaterialBanner createMaterialBanner(
   BuildContext context, {
   required StatusType type,
-  required Widget content,
-  required List<Widget> actions,
+  required String message,
+  required void Function() actionCallback,
 }) {
-  final margin = MediaQuery.of(context).viewPadding.top;
+  final theme = Theme.of(context);
+
+  final colorsChoices = getColors(theme.bannerTheme.backgroundColor,
+      theme.bannerTheme.contentTextStyle?.color, type, theme);
 
   return MaterialBanner(
     key: keyAuthenticatorBanner,
-    leading: Icon(type.icon),
-    padding: EdgeInsetsDirectional.only(
-      start: 16.0,
-      top: 2.0 + margin,
-      bottom: 4.0,
+    backgroundColor: colorsChoices.background,
+    leading: Icon(type.icon, color: colorsChoices.foreground),
+    content: Center(
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          Expanded(
+            child: Text(
+              message.trimRight(),
+              textAlign: TextAlign.left,
+              style: TextStyle(
+                color: colorsChoices.foreground,
+              ),
+            ),
+          ),
+        ],
+      ),
     ),
-    content: content,
-    actions: actions,
+    actions: [
+      IconButton(
+        onPressed: actionCallback,
+        icon: Icon(
+          Icons.close,
+          color: colorsChoices.foreground,
+        ),
+      ),
+    ],
   );
 }
 
 SnackBar createSnackBar(
   BuildContext context, {
   required StatusType type,
-  required Widget content,
+  required String message,
 }) {
   final theme = Theme.of(context);
 
+  final colorsChoices = getColors(theme.snackBarTheme.backgroundColor,
+      theme.snackBarTheme.contentTextStyle?.color, type, theme);
+
   // if contentTextStyle is null, colorScheme.surface is used as a fallback
   // since SnackBarThemeData.backgroundColor uses colorScheme.onSurface as a fall back
-  final foregroundColor =
+  // this is only used for dark/light mode for the icon color
+  final fallbackIconColor =
       theme.snackBarTheme.contentTextStyle?.color ?? theme.colorScheme.surface;
 
   return SnackBar(
     key: keyAuthenticatorBanner,
+    backgroundColor: colorsChoices.background,
     content: Row(
+      crossAxisAlignment: CrossAxisAlignment.center,
       children: [
-        Icon(
-          type.icon,
-          color: foregroundColor,
+        Icon(type.icon, color: fallbackIconColor),
+        const SizedBox(
+          width: 16,
         ),
-        const SizedBox(width: 16),
-        Expanded(child: content),
+        Expanded(
+          child: Text(
+            message.trimRight(),
+            textAlign: TextAlign.left,
```

Comment:
*nit:* this defaults to `TextAlign.start` which I think in this case has the same effect. You can probably just remove this.

Reply:
The comment applies the the banner as well.